lib/AST/SearchPathOptions.cpp

```diff
@@ -114,3 +115,40 @@ ModuleSearchPathLookup::searchPathsContainingFile(
                         const ModuleSearchPath *Rhs) { return *Lhs < *Rhs; });
   return Result;
 }
+
+static llvm::Error loadAndValidateVFSOverlay(
```

Comment:
Could you add a doc-comment to this method? If I understand correctly, it should be something like
```cpp
/// Loads a VFS YAML file located at \p File using \p BaseFS, creates an overlay file system from that definition and adds the overlay file system to \p OverlayFS.
```

Reply:
Sure, though to be clear it actually makes a RedirectingFileSystem :). At some point this function will be replaced entirely with one that I'll make in clang (technically have already made, but need other changes to get it in).